src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,321 @@
+import {
+  Component,
+  h,
+  Prop,
+  Element,
+  Event,
+  EventEmitter,
+  VNode,
+  Watch,
+  Method,
+  Host,
+  State
+} from "@stencil/core";
+import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
+import { connectLabel, disconnectLabel, getLabelText, LabelableComponent } from "../../utils/label";
+import { getSlotted } from "../../utils/dom";
+import { CSS, ERROR_MESSAGES, SLOTS } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+@Component({
+  tag: "calcite-textarea",
+  styleUrl: "textarea.scss",
+  shadow: true
+})
+export class Textarea implements FormComponent, LabelableComponent, LocalizedComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextareaElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When `true`, focuses the `textarea` element on page render. */
+  @Prop({ reflect: true }) autofocus: boolean;
+
+  /** When `true`, disables the component. */
+  @Prop({ reflect: true }) disabled: boolean;
+
+  /** Specifies the placeholder text for the input. */
+  @Prop({ reflect: true }) placeholder: string;
+
+  /** Whne `true`, the component's value can be read, but cannot be modified.  */
+  @Prop({ reflect: true }) readonly: boolean;
+
+  /** Specifies number or rows allowed. */
+  @Prop({ reflect: true }) rows: number;
+
+  /** Specifies number or columns allowed. */
+  @Prop({ reflect: true }) cols: number;
+
+  /** Specifies maximum number of characters allowed. */
+  @Prop({ reflect: true }) maxlength: number;
+
+  /** Specifies name of the component  */
+  @Prop({ reflect: true }) name: string;
+
+  /** Specifies the size of `textarea` component. */
+  @Prop({ reflect: true }) scale: "l" | "m" | "s" = "m";
+
+  /** Specifies wrapping mechanism for the text.  */
+  @Prop({ reflect: true }) wrap: "soft" | "hard" = "soft";
+
+  // @Prop() autocomplete: boolean;
```

Comment:
Whats this for?

Reply:
Good catch .this needs to be removed. I was trying to declare global attributes earlier and forgot to clean it.